src/main/java/org/opensearch/knn/index/query/iterators/ByteVectorIdsKNNIterator.java

```diff
@@ -72,20 +72,28 @@ public float score() {
     }
 
     protected float computeScore() throws IOException {
-        final byte[] vector = binaryVectorValues.getVector();
+        final byte[] vector = byteVectorValues.getVector();
         // Calculates a similarity score between the two vectors with a specified function. Higher similarity
         // scores correspond to closer vectors.
-        return spaceType.getKnnVectorSimilarityFunction().compare(queryVector, vector);
+
+        // The query vector of Faiss byte vector is a Float array because ScalarQuantizer accepts it as float array.
+        // Now, to compute the score between this query vector and each vector in KNNByteVectorValues we are casting this query vector into
+        // byte array.
```

Comment:
```suggestion
        // The query vector of Faiss byte vector is a Float array because ScalarQuantizer accepts it as float array.
        // To compute the score between this query vector and each vector in KNNByteVectorValues we are casting this query vector into byte array directly.
        // This is safe to do so because float query vector already has validated byte values. Do not reuse this direct cast at any other place
```